src/module.c

```diff
@@ -10909,6 +10931,11 @@ void moduleFireServerEvent(uint64_t eid, int subid, void *data) {
                 moduledata = data;
             } else if (eid == REDISMODULE_EVENT_CONFIG) {
                 moduledata = data;
+            } else if (eid == REDISMODULE_EVENT_KEY) {
+                moduledata = data;
+                RedisModuleKeyInfoV1 *ki = data;
+                if (ki->dbnum != -1)
+                    selectDb(ctx.client, ki->dbnum);
```

Comment:
```suggestion
                serverAssert(ki->dbnum != -1);
                selectDb(ctx.client, ki->dbnum);
```
Is ki->dbnum ever not set in this flow? I would rather assert since I assume it's always set.